### ISSUE TITLE
Consider collapsing the Light/Camera Orientation nodes into their parents

### ISSUE THREAD

**Issue by Clockwork-Muse** [KhronosGroup/glTF-Blender-IO#1328]
Currently, when lights and cameras are exported with +Y as the up axis (which is the default, and is mandated by the gltf spec), they are exported with an additional orientation "correction" node:
```json
nodes" : [
    {
        "extensions" : {
            "KHR_lights_punctual" : {
                "light" : 0
            }
        },
        "name" : "SomeLight_Orientation",
        "rotation" : [
            -0.7071067690849304,
            0,
            0,
            0.7071067690849304
        ]
    },
    {
        "children" : [
            0
        ],
        "name" : "SomeLight",
        "rotation" : [
            0.16907575726509094,
            0.7558803558349609,
            -0.27217137813568115,
            0.570947527885437
        ],
        "translation" : [
            4.076245307922363,
            5.903861999511719,
            -1.0054539442062378
        ]
    },
]
```
This can get annoying in some importers, including Blender's, because now there's an extra node in the tree, when there isn't one in the source file.  In Blender's case, it generates an empty (example from exporting and reimporting the default scene):
![image](https://user-images.githubusercontent.com/5569508/107555587-31cc5400-6b8c-11eb-993d-3d1be007af37.png)

**Desired solution**
Remove the use of a separate "correction" node, and combine the rotations to match the gltf spec in one node:
```json
"nodes" : [
    {
        "extensions" : {
            "KHR_lights_punctual" : {
                "light" : 0
            }
        },
        "name" : "SomeLight",
        // No idea what the values should actually be, but you get the idea
        "rotation" : [
            0.16907575726509094,
            0.7558803558349609,
            0.27217137813568115,
            0.570947527885437
        ],
        "translation" : [
            4.076245307922363,
            5.903861999511719,
            -1.0054539442062378
        ]
    },
]
```

**Additional notes**
I'm not sure why a separate orientation node was chosen in this case.  Exporting as Z-up is problematic, because attempting to re-import the file yields an odd orientation to the contents (because the importer naturally assumes the files should be Y-up).

Related to that, with the spec requiring +Y-up, should that option be removed from the exporter?

**Comment by donmccurdy** [comment]
This was discussed a bit in https://github.com/KhronosGroup/glTF-Blender-IO/issues/1171 and https://github.com/KhronosGroup/glTF-Blender-IO/issues/604#issuecomment-517387307  — I think we all agree that it would be nice to skip the extra node, but it gets complicated with animation.

**Comment by donmccurdy** [comment]
> Related to that, with the spec requiring +Y-up, should that option be removed from the exporter?

Related: https://github.com/KhronosGroup/glTF-Blender-IO/issues/348

**Comment by Clockwork-Muse** [comment]
@donmccurdy - Ah, so technically this one is a duplicate

**Comment by donmccurdy** [comment]
Perhaps ... will leave this open for at least a few days in case anyone has ideas we haven't tried yet for removing the extra node.

**Comment by scurest** [comment]
It's easy if everything is baked like https://github.com/KhronosGroup/glTF-Blender-IO/issues/1275#issuecomment-737656429.

**Comment by Alexeyt89** [comment]
Would it correct the situation if apply Y-Up correction to all exported Quaternions and Matrxies and at the same time apply inversed shift-rotation correction to children nodes to keep them where they should be? 
If do it for all nodes and animation frames, the mesh will be where it should be but all nodes will have switched axeses.
Here are 2 nodes get rotated and the mesh position stays the same:
![transformation](https://user-images.githubusercontent.com/16538686/114417417-40e67700-9bba-11eb-9e18-78c735688af8.png)
The same way it could be done for importing, only in a reversed way.

**Comment by scurest** [comment]
@Alexeyt89 Yeeessss, I think so. That's already implemented in the importer. It's how it imports cameras without an "orientation" node, and also how it switches bone directions around (see #945). I called that a "local rotation" because it doesn't affect the children's world matrix.

```
(new parent TRS) = (original parent TRS) (local rotation)
(new child TRS) = (local rotation).inverted() (original child TRS)
(new child world) = ... (new parent TRS) (new child TRS)
                  = ... (original parent TRS) (original child TRS)
```

**Comment by julienduroure** [comment]
Fixed by 69746b2b31eeee080d5dfd3c74f1e02742795ed2
